client/my-sites/patterns/index.node.tsx

```diff
@@ -91,6 +91,11 @@ function fetchCategoriesAndPatterns( context: RouterContext, next: RouterNext )
 		} );
 }
 
+function excludeSearchFromCanonicalUrl( context: RouterContext, next: RouterNext ) {
+	context.excludeSearchFromCanonicalUrl = true;
+	next();
+}
+
```

Comment:
I would suggest moving that middleware to `client/controller/localized-links.js`, so it's closer to the code that actually uses that particular context property. It's also generic enough that it could be used easily in other entry points.